kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -368,7 +368,9 @@ private void handleTxn(ColumnVector txnVector) {
   }
 
   private List<FileStatus> getWinningCommitFiles(Engine engine) {
-    String firstWinningCommitFile = deltaFile(snapshot.getLogPath(), snapshot.getVersion() + 1);
+    // TODO doesn't current impl mean we always try to resolve conflicts from readSnapshot version
+    //  again? instead of iteratively on only the newest commit
+    String firstWinningCommitFile = deltaFile(transaction.getLogPath(), attemptVersion);
```

Comment:
good catch!
- attempt 1: we want to write `10.json`. We conflict and see that `10.json` to `15.json` exist. We must check against 10.json to 15.json
- attempt 2: we want to write `16.json`. We conflict and see that `16.json` to `20.json` exist. We must check against `16` to `20`. We should NOT check against 10 to 15, too